api/controllers/PostController.js

```diff
@@ -392,6 +400,36 @@ module.exports = {
       }
     })
     .then(() => res.ok({}), res.serverError)
-  }
+  },
+
+  checkFreshnessForCommunity: function (req, res) {
+    var foundPosts
+    var resolve = value => foundPosts = value
```

Comment:
the way that seems most straightforward to me is that you send as parameters the list of post ids you have along with the updated_at timestamp for each. then all you need to do on the server side is `Post.query().where('id', 'in', thoseIds).filter(p => p.get('updated_at') > originalUpdatedAts[p.id])` (pseudocode, but i hope you get the idea) and return `true` if that list is non-empty.

then you avoid using `findPosts` at all, because you don't need to build the original query from scratch, and you don't need any of the presentation logic.